### PR TITLE
Fix crash when closing dialogs with X or Cancel button.

### PR DESCRIPTION
**Pull request by Slion**

### CODE REVIEW

**Comment by sthalik** [comment]
Drop me the stack trace please.

**Comment by sthalik** [comment]
This isn't the proper fix but the crash needs to be gone, of course.

File: api/plugin-api.cpp [hunk 1]
```diff
@@ -5,15 +5,13 @@
 namespace plugin_api::detail {
 
 BaseDialog::BaseDialog() = default;
-void BaseDialog::closeEvent(QCloseEvent*)
+
+void BaseDialog::closeEvent(QCloseEvent* aEvent)
 {
-    if (isVisible())
-    {
-        hide();
```
Comment (sthalik) [inline api/plugin-api.cpp]: These two lines can't be removed due to other core functionality depending on it.

**Comment by Slion** [comment]
I did not get a proper stack trace, it crashes somewhere in Qt. I guess I could try compiling Qt myself but it could be a while until I get there.

**Comment by sthalik** [comment]
Is there some easy way to reproduce it?

**Comment by Slion** [comment]
> Is there some easy way to reproduce it?

It pretty much always occur when I debug my x64 build. Just open a tracker settings dialog and close it using cancel or X button. Though yesterday I noticed it did not occur, at least one time I clicked on cancel. I thought you may have fixed it already but left it at that since I was working on something else.

**Comment by sthalik** [comment]
Can you reproduce it with any tracker?

**Comment by Slion** [comment]
> Can you reproduce it with any tracker?

I did not try them all but I pretty sure Point Tracker had that issue too.

**Comment by Slion** [comment]
Actually it looks like it's just the X button. Any dialog I tried thus far. Input, Output, Filter settings.
`Exception thrown at 0x00007FFBB58DDF9D (Qt5Guid.dll) in opentrack.exe: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.`

**Comment by sthalik** [comment]
Can you reproduce it with the snapshots? These have pdb's in the debug directory.

**Comment by Slion** [comment]
> Can you reproduce it with the snapshots? These have pdb's in the debug directory.

You mean the releases? I think it works fine there for some reason.

**Comment by sthalik** [comment]
I'm unable to reproduce it with Valgrind either.

**Comment by sthalik** [comment]
Can you confirm if there are any crashes now? There's improved `run-in-thread-sync` usage in the options system now.

**Comment by sthalik** [comment]
Closing due to having been fixed through f5f3bd69fa64834b555ec8e224271756f017527e , b29b8e48fe59f44fd28de5f6879730bcdf34a9dd, and d3ffafd227ab99e8673ea3edcf31de9b9e69fffa.

**Pull request closed without merging**